ee/desktop/server/server.go

```diff
@@ -0,0 +1,71 @@
+package server
+
+import (
+	"context"
+	"net/http"
+	"os"
+
+	"fyne.io/systray"
+	"github.com/kolide/launcher/ee/desktop"
+)
+
+var shutdownChan = make(chan struct{})
+
+// Start spins up a server that listens to a unix socket or named pipe for windows.
+//
+// On macos & linux you can curl this with:
+//
+//	curl --unix-socket /tmp/<pid>_launcher_desktop.sock ./<endpoint>
+//
+// For example:
+//
+//	curl --unix-socket /tmp/1234_launcher_desktop.sock ./shutdown
+//
+// On windows the socket path is \\.\pipe\<pid>_launcher_desktop.sock, but curl
+// doesn't seem to support named pipes on windows.
+func Start() error {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/shutdown", shutdown)
+
+	server := http.Server{
+		Handler: mux,
+	}
+
+	socketPath := desktop.DesktopSocketPath(os.Getpid())
+
+	if err := os.RemoveAll(socketPath); err != nil {
+		//TODO: log this
+	}
+
+	listener, err := listener(socketPath)
+	if err != nil {
+		//TODO: log this
+	}
+
+	go func() {
+		if err := server.Serve(listener); err != nil {
+			// TODO: log this
+			shutdownChan <- struct{}{}
+		}
+	}()
+
+	<-shutdownChan
+
+	if err := os.RemoveAll(socketPath); err != nil {
+		//TODO: log this
+	}
+
+	if err := server.Shutdown(context.Background()); err != nil {
+		//TODO: log this
+	}
+
+	systray.Quit()
```

Comment:
wonder if this pattern is correct .... so the server be calling `systray.Quit()` or should it accept a channel as a parameter and send on the channel when it gets a shut down signal and let the calling process deal with it? Something else?

Reply:
This feels weird. And maybe dovetails with your package comment. If the server is over here, then it shouldn't be calling `systray.Quit()`. It should have something like a rungroup it triggers shutdown for.

Which makes me wonder about moving a bunch of this into main. Main would create a rungroup or shutdown channel, and then start a server using it. (launcher kinda does that)